### PR TITLE
Prevent duplicate pings

### LINKED ISSUE #1: Fix duplicate pings
See https://github.com/robotland/gjtorikian-made-me-do-it/pull/1

I think there are two changes required here:

1. Only trigger on `pull_request.synchronize` event (meaning code changed)
2. Don't ping someone if they've already been pinged in a thread.

### PR DESCRIPTION
**Pull request by lee-dohm**
Fixes #1 

* Add test infrastructure
* Refactor to make more testable
* Convert to using async/await instead of explicit `Promises`
* Add _lots_ of tests

and of course:

* Change to trigger only on `pull_request.synchronize`
* Only ping owners that haven't already been pinged

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,11 +14,20 @@
...
     "esnext": true,
-    "space": true
+    "space": true,
+    "envs": [
+      "node",
+      "mocha"
+    ],
+    "rules": {
+      "camelcase": 1
+    }
```
Comment (bkeepers) [inline package.json:30]: I'm starting to wonder if these rules should be pulled out into a common config for all probot plugins somewhere.

File: lib/owner-notifier.js [hunk 2]
```diff
@@ -13,54 +14,62 @@ class OwnerNotifier {
...
-      return this.github.issues.createComment(Object.assign({
-        number: this.event.payload.number,
-        body: '/cc ' + changes.owners.join(' ')
-      }, this.repo));
-    }
+  async getOwners() {
+    const options = Object.assign({path: 'OWNERS'}, this.repo);
+    const data = await this.github.repos.getContent(options);
+
+    return ownersFile(new Buffer(data.content, 'base64').toString());
```
Comment (bkeepers) [inline lib/owner-notifier.js:46]: I think some way of easily fetching content is due.  I'd love to do `context.content(path)`, but it needs an authenticated client, which is making me wonder about adding `context.github`.

**Review by bkeepers — APPROVED** [review]
Looks _really_ great.  I just commented on a few things that I'm starting to see in several plugins, but nothing to change here now!

File: package.json (hunk 1 above)
Comment (lee-dohm) [reply package.json:30]: Yeah, that would make a lot of sense :+1:

Or perhaps a probot plugin generator that drops a standard template like the [Atom package-generator package](https://github.com/atom/package-generator)? Maybe just make it a probot command-line command?

File: lib/owner-notifier.js (hunk 2 above)
Comment (lee-dohm) [reply lib/owner-notifier.js:46]: It only needs an authenticated client for private content. So we could at least start with `context.content(path)` for unauthenticated content and provide some other mechanism for authenticated stuff?

**Comment by lee-dohm** [comment]
I'm going to manually test this tomorrow to ensure that it works as expected and I didn't blow anything up. If that goes well, I'll merge this :+1:

**Comment by lee-dohm** [comment]
Looks like we need to also respond to `pull_request.opened`. I'm getting on an :airplane: right now though, so I'll have to fix this when I land :grinning:

**Comment by bkeepers** [comment]
:zap: nice work!

**Pull request merged**